Source/JavaScriptCore/jit/AssemblyHelpers.h

```diff
@@ -1483,6 +1483,27 @@ class AssemblyHelpers : public MacroAssembler {
         
         done.link(this);
     }
+
+    void branchConvertDoubleToInt52(FPRegisterID srcFPR, RegisterID destGPR, JumpList& failureCases, RegisterID scratch1GPR, FPRegisterID scratch2FPR)
+    {
+        JumpList doneCases;
+
+        truncateDoubleToInt64(srcFPR, destGPR);
+        convertInt64ToDouble(destGPR, scratch2FPR);
+        failureCases.append(branchDouble(DoubleNotEqualOrUnordered, srcFPR, scratch2FPR));
+        auto isZero = branchTest64(Zero, destGPR);
+        move(TrustedImm64(0xfff0000000000000ULL), scratch1GPR);
+        add64(destGPR, scratch1GPR);
+        urshift64(TrustedImm32(53), scratch1GPR);
+        failureCases.append(branch64(BelowOrEqual, scratch1GPR, TrustedImm32(2046)));
```

Comment:
This check implementation seems non-obvious, at least to me - maybe we could have a short comment here explaining the logic? Or maybe a pointer to the right file if this is already explained elsewhere.

Reply:
Added.